### PR TITLE
Framework: Update get_dependencies.sh script to use open-sourced repo

### PR DESCRIPTION
**Pull request by achauphan**
@trilinos/framework 

## Motivation

GenConfig has released its repositories on GitHub. Our Framework scripts should utilizes these repositories instead of the versions hosted on internal repositories. This effort also contributes to the eventual release of our Framework containers.

## Testing

Manually tested this script on SRN and in our containerized environment and was able to load existing configurations.

Further testing should be observed in the CI.

### CODE REVIEW

File: packages/framework/get_dependencies.sh [hunk 1]
```diff
@@ -3,7 +3,8 @@ ini_file_option=$1
 script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" &> /dev/null && pwd)"
 
 # Data that needs to be updated when GenConfig changes!
-genconfig_sha1=924a08af66f0a0573b5dd1128179731489339aec
+genconfig_sha1=HEAD
```
Comment (sebrowne) [inline packages/framework/get_dependencies.sh:6]:
```suggestion
genconfig_sha1=e24b800d6673085850fb612d17ad44744e148464
```

Comment (sebrowne) [reply packages/framework/get_dependencies.sh:6]: Use the SHA1 of the current main branch tip, that way we can upgrade it via PR here and know it's still working.

File: packages/framework/get_dependencies.sh [hunk 2]
```diff
@@ -59,7 +60,7 @@ function tril_genconfig_clone_or_update_repo() {
   if [[ "${has_submodules}" == "has-submodules" ]] ; then
     echo
     echo "STATUS: ${sub_dir}: Update submodules"
-    cmd="git submodule update --force --init --recursive"
+    cmd="git submodule update --force --init"
```
Comment (sebrowne) [inline packages/framework/get_dependencies.sh:63]: I think you still want the`--recursive` here.

**Review by sebrowne — CHANGES_REQUESTED** [review]

**Comment by achauphan** [comment]
Retesting, gcc-serial and gcc-openmpi-openmp jobs failed after not being able to source sems modules.

**Review by sebrowne — APPROVED** [review]

**Pull request merged**